### ISSUE TITLE
[cmd/telemetrygen] Bad default argument values for HTTP exporter mode

### ISSUE THREAD

**Issue by jmsnll** [open-telemetry/opentelemetry-collector-contrib#26999]
### Component(s)

cmd/telemetrygen

## Description

The default behaviour of telemetrygen when trying to use HTTP exporters is broken out of the box.

1. `otlp-http-url-path` is current set in `internal/common/config.go` file with a default value of `/v1/metrics` so it is pointing toward the wrong endpoint when trying to use the `traces` sub-command
2. `otlp-http` is used to indicate weather to use gRPC or HTTP exporters but `otlp-endpoint` always has a default value pointing toward the gRPC receiver port `4317`

## Steps to Reproduce

Have a locally running collector with HTTP and gRPC receivers enabled, then run telemetrygen to send traces using HTTP with: `telemetrygen traces --otlp-http --otlp-insecure` 

## Expected Result

1. Because we are sending traces `otlp-http-url-path` has a value of `/v1/traces` by default
2. Because we have set the `otlp-http` flag and have not set a value for `otlp-endpoint` we send traces to `localhost:4318`

```
$ telemetrygen traces --otlp-http --otlp-insecure 
INFO    traces/traces.go:59     starting HTTP exporter
INFO    traces/traces.go:120    generation of traces isn't being throttled
INFO    traces/worker.go:92     traces generated        {"worker": 0, "traces": 1}
INFO    traces/traces.go:80     stop the batch span processor
INFO    traces/traces.go:70     stopping the exporter
```

## Actual Result

We attempt to send traces via HTTP to the wrong endpoint (`/v1/metrics`) and the incorrect default port for the HTTP receiver `:4317`

```
$ telemetrygen traces --otlp-http --otlp-insecure
INFO    traces/traces.go:59     starting HTTP exporter
INFO    traces/traces.go:120    generation of traces isn't being throttled
INFO    traces/worker.go:92     traces generated        {"worker": 0, "traces": 1}
INFO    traces/traces.go:80     stop the batch span processor
traces export: Post "http://localhost:4317/v1/metrics": net/http: HTTP/1.x transport connection broken: malformed HTTP response "\x00\x00\x06\x04\x00\x00\x00\x00\x00\x00\x05\x00\x00@\x00"
INFO    traces/traces.go:70     stopping the exporter
```

## Proposed Solution

- Remove the declaration of `otlp-http-url-path` from the common flags and have one in each of the `internal/traces/config.go` and `internal/metrics/config.go` module with the appropriate default value.
- Check if `otlp-http` has been set and override `otlp-endpoint` to `localhost:4318` if it has it's default value of `localhost:4317` 

### Collector version

v0.85.0

**Comment by mx-psi** [comment]
Thanks for the report @jmsnll, assigning P1 since this is important to fix for alignment with the OTLP spec

**Comment by jmsnll** [comment]
My pleasure! I'm happy to open a PR for this also if my proposed solution looks sound

**Comment by mx-psi** [comment]
> My pleasure! I'm happy to open a PR for this also if my proposed solution looks sound

The proposed solution makes sense to me, feel free to open a PR! :)

**Comment by mx-psi** [comment]
Fixed by #27007
